File: src/basic_study/STL_stack.java
package basic_study;

import java.util.Stack;

public class STL_stack {
	public static void main(String[] args) {
		// 在Java中，我们用Deque可以实现Stack的功能
		/**
		 * 把元素压栈：push(E)/addFirst(E)； 把栈顶的元素“弹出”：pop(E)/removeFirst()；
		 * 取栈顶元素但不弹出：peek(E)/peekFirst()。
		 */
		Stack<Integer> lis = new Stack<>();
		lis.push(2);
		lis.push(4);
		System.out.println(lis.peek());
		System.out.println(lis.pop());
		System.out.println(lis.peek());

	}
}

File: c++/base_study/STL_stack.cpp
//c++ stl栈stack的头文件为: 
#include <stack> 
//c++ stl栈stack的成员函数介绍
//
//操作 比较和分配堆栈
//
//empty() 堆栈为空则返回真
//
//pop() 移除栈顶元素
//
//push() 在栈顶增加元素
//
//size() 返回栈中元素数目
//
//top() 返回栈顶元素
//
using namespace std;  
int main(void)  
{  
    stack<double>s;//定义一个栈  
    for(int i=0;i<10;i++)  
        s.push(i);  
    while(!s.empty())  
    {  
        printf("%lf\n",s.top());  
        s.pop();  
    }  
    cout<<"栈内的元素的个数为："<<s.size()<<endl;  
    return 0;  
}
